apply stylelint auto fixes

christian-byrne opened the pull request:
## Summary

Applied stylelint auto-fixes and resolved manual CSS issues across 25 files to achieve full compliance with stylelint rules.

## Changes

- **What**: Auto-fixed 68 CSS issues (legacy color functions, font-weight keywords, shorthand properties, pseudo-element notation) and manually resolved 6 remaining issues (duplicate selectors, vendor prefix duplicates, font fallbacks, Vue v-bind whitelisting)
- **Config**: Disabled `no-descending-specificity` rule (43 warnings require architectural CSS refactor)

## Review Focus

Verify no visual regressions from modernized CSS syntax:
- Modern [color function notation](https://www.w3.org/TR/css-color-4/#funcdef-rgb): `rgba(0, 0, 0, 0.5)` → `rgb(0 0 0 / 50%)`
- Numeric font weights: `bold`/`normal` → `700`/`400`
- Pseudo-element double colons: `:before` → `::before

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-5940-apply-stylelint-auto-fixes-2846d73d365081ee8031c212a69a4bd4) by [Unito](https://www.unito.io)

christian-byrne committed:
fix: apply stylelint auto-fixes

Auto-fix 68 CSS issues found by stylelint:
- Convert legacy color functions (rgba() → rgb())
- Normalize font weights (bold/normal → 700/400)
- Use shorthand properties (top/right/bottom/left → inset)
- Add url() wrapper to @import statements
- Remove redundant shorthand values
- Fix pseudo-element notation (: → ::)

Remaining 45 issues require manual review:
- 44 descending specificity warnings
- 1 duplicate selector

christian-byrne committed:
fix: resolve remaining stylelint issues

- Disable no-descending-specificity (43 warnings, requires CSS refactor)
- Fix 2 duplicate selectors (merge into single declaration)
- Remove duplicate vendor prefix fallbacks (user-select, mask-image)
- Add font fallbacks for primeicons (sans-serif)
- Whitelist Vue v-bind() CSS function

christian-byrne committed:
chore: clean up knip configuration

Remove stylelint from ignoreDependencies and ignoreBinaries now that it's actively used.
Add tw-animate-css and tailwindcss to ignoreDependencies as they're used in CSS @import statements which knip doesn't detect.

DrJKL commented on knip.config.ts:
Ah, I think you'll need to update this to account for `url()`

christian-byrne replied:
https://github.com/Comfy-Org/ComfyUI_frontend/pull/5940/commits/96d33f1cc2ce6192b404ce28b8d99b702ef34cfb

DrJKL committed:
fix: Change stylelint import-notation to string

DrJKL committed:
cleanup: unignore tailwind imports

christian-byrne committed:
fix: address PR review comments - revert shorthand CSS properties

- Remove `declaration-block-no-redundant-longhand-properties` stylelint rule
  to avoid browser compatibility issues with modern CSS shorthands
- Revert `flex-flow` and `place-content` back to longhand properties
  in NodePreview.vue for better browser support
- Revert `inset: 0` back to longhand `top/left/right/bottom` in
  TreeExplorer.vue for browser compatibility
- Update knip.config.ts CSS compiler to properly handle `url()` wrappers
  in @import statements

christian-byrne commented on .stylelintrc.json:
```diff
@@ -35,7 +36,7 @@
     "declaration-block-no-duplicate-properties": true,
     "block-no-empty": true,
-    "no-descending-specificity": true,
+    "no-descending-specificity": null,
```
It's because it would require its own PR with a more care.

christian-byrne commented on src/components/node/NodePreview.vue:
```diff
@@ -252,10 +251,8 @@ const truncateDefaultValue = (value: any, charLimit: number = 32): string => {
-  flex-wrap: nowrap;
-  align-content: flex-start;
-  justify-content: flex-end;
+  flex-flow: row-reverse nowrap;
```
Reverted this rule

christian-byrne commented on src/components/common/TreeExplorer.vue:
```diff
@@ -263,10 +263,7 @@ defineExpose({
-  left: 0;
-  right: 0;
-  bottom: 0;
+  inset: 0;
```
Reverted this rule

christian-byrne commented:
I removed the *declaration-block-no-redundant-longhand-properties* rule and reverted the relevant fixes - according to Claude, it forces us to use shorthand which is sometimes not super well supported and the tradeoff does not seem worth it.

christian-byrne commented on src/components/node/NodePreview.vue:
```diff
@@ -252,10 +251,8 @@ const truncateDefaultValue = (value: any, charLimit: number = 32): string => {
-  align-content: flex-start;
-  justify-content: flex-end;
+  flex-flow: row-reverse nowrap;
+  place-content: flex-start flex-end;
```
Removed that rule

DrJKL commented:
> I removed the _declaration-block-no-redundant-longhand-properties_ rule and reverted the relevant fixes - according to Claude, it forces us to use shorthand which is sometimes not super well supported and the tradeoff does not seem worth it.

I _generally_ prefer the shorthand, but we can always add rules incrementally later 🙂

christian-byrne committed:
fix: revert remaining shorthand CSS properties for browser compatibility

- Revert all remaining `inset:` shorthands to longhand in design-system,
  SidebarHelpCenterIcon, menu.css, and PackCard
- Revert `flex-flow` and `place-content` to longhand in design-system

DrJKL commented on packages/design-system/src/css/style.css:
```diff
@@ -746,7 +746,10 @@ button.comfy-queue-btn {
+    top: 0 !important;
+    bottom: 0 !important;
+    left: auto !important;
+    right: 0 !important;
```
The importants still make me sad
